src/main/java/org/openlmis/buq/web/buq/BottomUpQuantificationController.java

```diff
@@ -0,0 +1,216 @@
+import org.openlmis.buq.exception.NotFoundException;
+import org.openlmis.buq.i18n.MessageKeys;
+import org.openlmis.buq.repository.buq.BottomUpQuantificationRepository;
+import org.openlmis.buq.service.buq.BottomUpQuantificationService;
+import org.openlmis.buq.util.Pagination;
+import org.openlmis.buq.web.BaseController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+/**
+ * Controller used to expose Bottom-Up Quantifications via HTTP.
+ */
+@Controller
+@RequestMapping(BottomUpQuantificationController.RESOURCE_PATH)
+@Transactional
+public class BottomUpQuantificationController extends BaseController {
+
+  public static final String RESOURCE_PATH = API_PATH + "/bottomUpQuantifications";
+
+  @Autowired
+  private BottomUpQuantificationRepository bottomUpQuantificationRepository;
+
+  @Autowired
+  private BottomUpQuantificationService bottomUpQuantificationService;
+
+  /**
+   * Retrieves all bottom-up quantifications. Note that an empty collection rather than a 404
+   * should be returned if no bottom-up quantifications exist.
+   */
+  @GetMapping
+  @ResponseStatus(HttpStatus.OK)
+  @ResponseBody
+  public Page<BottomUpQuantificationDto> getAllBottomUpQuantifications(Pageable pageable) {
+    Page<BottomUpQuantification> page = bottomUpQuantificationRepository.findAll(pageable);
+    List<BottomUpQuantificationDto> content = page
+        .getContent()
+        .stream()
+        .map(this::buildDto)
+        .collect(Collectors.toList());
+    return Pagination.getPage(content, pageable, page.getTotalElements());
+  }
+
+  /**
+   * Retrieves the specified bottom-up quantification.
+   */
+  @GetMapping(value = "/{id}")
+  @ResponseStatus(HttpStatus.OK)
+  @ResponseBody
+  public BottomUpQuantificationDto getSpecifiedBottomUpQuantification(
+      @PathVariable("id") UUID id) {
+    BottomUpQuantification buq = bottomUpQuantificationRepository.findById(id).orElseThrow(
+        () -> new NotFoundException(MessageKeys.ERROR_BOTTOM_UP_QUANTIFICATION_NOT_FOUND));
```

Comment:
e.g. here in my opinion we should call the `bottomUpQuantificationService.findById(id)` and it should handle this business logic

Reply:
I stuck to the OpenLMIS convention where both Repository and Service are injected into the controllers as needed.
For simple methods like "get single" the repository is used directly, while for methods requiring more complex logic services are used.